pkg/awsds/sessions.go

```
@@ -26,12 +29,42 @@ type envelope struct {
 type SessionCache struct {
 	sessCache     map[string]envelope
 	sessCacheLock sync.RWMutex
+	authSettings  *AuthSettings
 }
 
 // NewSessionCache creates a new session cache
 func NewSessionCache() SessionCache {
+	authSettings := readAuthSettingsFromEnvironmentVariables()
 	return SessionCache{
-		sessCache: map[string]envelope{},
+		sessCache:    map[string]envelope{},
+		authSettings: authSettings,
+	}
+}
+
+func readAuthSettingsFromEnvironmentVariables() *AuthSettings {
+	allowedAuthProviders := []string{}
+	providers := os.Getenv("ALLOWED_AUTH_PROVIDERS")
+	for _, authProvider := range strings.Split(providers, ",") {
+		authProvider = strings.TrimSpace(authProvider)
+		if authProvider != "" {
+			allowedAuthProviders = append(allowedAuthProviders, authProvider)
+		}
+	}
+
+	if len(allowedAuthProviders) == 0 {
+		allowedAuthProviders = []string{"default", "keys", "credentials"}
+		plog.Warn("could not find allowed auth providers. falling back to 'default, keys, credentials'")
```

this is not ideal...I'd like to return an error, but that would break the existing api of creating sessions.

---

if we really want things to fail hard if we are missing expected environment variables, then we should create a common function and call it in the startup method of the various aws methods

Essentially add a one line function here:
https://github.com/grafana/iot-sitewise-datasource/blob/main/pkg/main.go#L13